packages/ui/src/components/Floating/Floating.tsx

```diff
@@ -59,10 +60,17 @@ export const Floating: FC<FloatingProps> = ({
   theme,
   trigger = "hover",
   minWidth,
+  isOpen = null,
   ...props
 }) => {
   const arrowRef = useRef<HTMLDivElement>(null);
-  const [open, setOpen] = useState(false);
+  const [innerOpen, setOpen] = useState(false);
+
+  const open = useMemo(() => {
```

Comment:
no need to use `useMemo()` here since its not a heavy calculation or any iteration, we can just stick to a more simplified version:

```tsx
  const [_open, setOpen] = useState(false);

  const open = isOpen ?? _open;
```